client/my-sites/people/delete-user/index.jsx

```diff
@@ -297,9 +307,13 @@ class DeleteUser extends React.Component {
 
 export default localize(
 	connect(
-		state => ( {
-			currentUser: getCurrentUser( state ),
-		} ),
+		( state, { siteId, user: { ID: userId } } ) => {
+			const externalContributors = ( siteId && requestExternalContributors( siteId ).data ) || [];
+			return {
+				currentUser: getCurrentUser( state ),
+				isExternalContributor: externalContributors.includes( userId ),
```

Comment:
this is probably fine but we should note that we're closing over valid and expected data states which will likely trigger some unexpected UI changes/jank

in cases like these where I don't want to pull into the component all of the logic for determining API status I like to create a new interface just for the data point I'm concerned about. think about what this component intends to show and then lock-in on an interface that can describe that (and which we can easily supply in a test function)

```js
const getRelationship = ( externalContributors, userId ) => {
	if ( externalContributors.data ) {
		return externalContributors.data.includes( userId )
			? 'external-contributor'
			: 'contributor';
	}

	return 'pending';
}

const mapStateToProps = ( state, { siteId, user: { ID: userId } } ) => {
	const externalContributors = siteId ? requestExternalContributors( siteId ) : httpData.empty;

	return {
		currentUser: getCurrentUser( state ),
		relationship: getRelationship( externalContributors, userId ),
	};
}
```

Now we have a tri-state value we can use to control things like the activation of the delete button…
```ts
<Button
	borderless
	className="…"
	onClick={ this.removeUser }
	disabled={ this.props.relationship === 'pending' }
/>
```

Now our UI can behave in a controlled manner while the data loads plus we can test it trivially…
```ts
import { DeleteUser } from '../';

it( 'should prevent deleting a user until the external contributor data is available', () => {
	const wrapper = mount( <DeleteUser … relationship="pending" /> );
	wrapper.find( 'whateverButton' ).simulate( 'click' );
	expect( deleteUser ).toNotHaveBeenCalled();
} );
```

^^^ that test code is wrong on so many levels but I hope it conveys the idea. we can specify in our tests the expected ways we want people to use this component and then demonstrate how they should behave in the various states of data fetching and handling. in this case I've collapsed the failure state so if the API call fails for any reason the button just stays disabled. there are more robust ways of responding to such a failure.